src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx

```diff
@@ -118,6 +124,82 @@ describe("Create New Cql Library Component", () => {
     expect(mockPush).toHaveBeenCalledWith("/example");
   });
 
+  it("should render the model options when clicked", async () => {
+    render(<CreateNewCqlLibrary />);
+    const modelDropdown = screen.getByRole("button", {
+      name: /select a model/i,
+    });
+    userEvent.click(modelDropdown);
+    const qiCoreOption = screen.getByText("QI-Core");
+    expect(qiCoreOption).toBeInTheDocument();
+  });
+
+  it("should update the dropdown with the selected option", async () => {
+    render(<CreateNewCqlLibrary />);
+    const modelDropdown = screen.getByRole("button", {
+      name: /select a model/i,
+    });
+    userEvent.click(modelDropdown);
+    const qiCoreOption = screen.getByText("QI-Core");
+    expect(qiCoreOption).toBeInTheDocument();
+    userEvent.click(qiCoreOption);
+    const qiCore = await screen.findByText("QI-Core");
+    expect(qiCore).toBeInTheDocument();
+    const qiCoreButton = screen.getByRole("button", { name: /qi-core/i });
+    expect(qiCoreButton).toBeInTheDocument();
+  });
+
+  it("should display an error when model is not selected", async () => {
+    // jest.setTimeout(15000);
+    render(<CreateNewCqlLibrary />);
+    const input = screen.getByTestId(
+      "cql-library-name-text-field"
+    ) as HTMLInputElement;
+    userEvent.type(input, "TestingLibraryName12");
+    expect(input.value).toBe("TestingLibraryName12");
+    const modelDropdown = screen.getByRole("button", {
+      name: /select a model/i,
+    });
+    expect(modelDropdown).toBeInTheDocument();
+    userEvent.click(modelDropdown);
+    const qiCoreOption = screen.getByText("QI-Core");
+    expect(qiCoreOption).toBeInTheDocument();
+    userEvent.type(modelDropdown, "{esc}");
+    userEvent.dblClick(input);
+    await waitFor(
+      () => {
+        expect(
+          screen.getByText("A CQL library model is required.")
+        ).toBeInTheDocument();
+      },
+      { timeout: 5000 }
+    );
+    // const errorMessage = await screen.findByText(
+    //   "A CQL library model is required.", null, { timeout: 2000 }
+    // );
+    const saveButton = screen.getByTestId("create-new-cql-library-save-button");
+    expect(saveButton).toBeDisabled();
+    // expect(errorMessage).toBeInTheDocument();
+  }, 15000);
+
+  // it.only("should have create button disabled until model is selected", async () => {
+  //   render(<CreateNewCqlLibrary />);
+  //   const input = screen.getByTestId(
+  //     "cql-library-name-text-field"
+  //   ) as HTMLInputElement;
+  //   userEvent.type(input, "TestingLibraryName12");
+  //   expect(input.value).toBe("TestingLibraryName12");
+  //   const modelDropdown = screen.getByRole("button", {
+  //     name: /select a model/i,
+  //   });
+  //   expect(modelDropdown).toBeInTheDocument();
+  //   screen.debug();
+  //   const saveButton = screen.getByTestId("create-new-cql-library-save-button");
+  //   await waitFor(() => {
+  //     expect(saveButton).toBeDisabled();
+  //   });
```

Comment:
Why is this test case commented ? I think its good to have, I missed it initially. good catch.

Reply:
I ended up just including this assertion in the test above

Reply:
I added another test (which took forever cause I needed a waitfor for some reason to get past one expect) to take the place of this